linux_os/guide/system/network/network-nftables/set_nftables_base_chain/bash/shared.sh

```diff
@@ -0,0 +1,55 @@
+# platform = multi_platform_sle
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+#Name of the table
+{{{ bash_instantiate_variables("var_nftables_table") }}}
+#Familiy of the table 
+{{{ bash_instantiate_variables("var_nftables_family") }}}
+#Name(s) of base chain
+{{{ bash_instantiate_variables("var_nftables_base_chain_names") }}}
+#Type(s) of base chain
+{{{ bash_instantiate_variables("var_nftables_base_chain_types") }}}
+# Hooks for base chain
+{{{ bash_instantiate_variables("var_nftables_base_chain_hooks") }}}
+#Priority
+{{{ bash_instantiate_variables("var_nftables_base_chain_priorities") }}}
+#Policy 
+{{{ bash_instantiate_variables("var_nftables_base_chain_policies") }}}
+
+#Transfer some of strings to arrays
+IFS="," read -r -a  names <<< "$var_nftables_base_chain_names"
+IFS="," read -r -a  types <<< "$var_nftables_base_chain_types"
+IFS="," read -r -a  hooks <<< "$var_nftables_base_chain_hooks"
+IFS="," read -r -a  priorities <<< "$var_nftables_base_chain_priorities"
+IFS="," read -r -a  policies <<< "$var_nftables_base_chain_policies"
+
+IS_TABLE_EXIST=$(nft list tables)
+if [ -z "$IS_TABLE_EXIST" ]
+then
+  # We create a table and add chains to it 
+  nft create table "$var_nftables_family" "$var_nftables_table"
+  num_of_chains=${#names[@]}
+  for ((i=0; i < num_of_chains; i++))
+  do
+   chain_to_add="add chain $var_nftables_family $var_nftables_table ${names[$i]} { type ${types[$i]} hook ${hooks[$i]} priority ${priorities[$i]} ; policy ${policies[$i]} ; }"
+   mycommand="nft '$chain_to_add'"
+   eval $mycommand
+  done    
+else
```

Comment:
This is not working when there are more than one table in the system. Also, even after fixing this, I don't think is always correct to assume the first table should be used. So, this approach is too fragile in my opinion. Maybe it would be better and simpler to not try to find the most appropriated table, but ensure the remediation uses only the specific values from the table and family variables.

Reply:
Hello @marcusburghardt 
Your proposal does not work. It is not possible to check for the existence of a specific table which belongs to a specific table's family. On my test machine I have:

sles-15-sp2:~ # nft flush ruleset

sles-15-sp2:~ # nft list table inet
Error: syntax error, unexpected newline, expecting string
list table inet
               ^
sles-15-sp2:~ # nft list table inet filter
Error: Could not process rule: Table 'filter' does not exist
list table inet filter
^^^^^^^^^^^^^^^^^^^^^^^
sles-15-sp2:~ # 

Only ntf list tables works
sles-15-sp2:~ # nft list tables
sles-15-sp2:~ # 



Have a nice day
Rumen

Reply:
The example you used `nft list table inet filter` is expected. If that specific table and family don't exist, the command will return error. Otherwise it will return success. So you only need to treat this behavior in the Playbook. But you can't assume the `nft list table` is enough to ensure the `inet filter` table exists.

Reply:
It is more precise now. Thanks